magento/pwa-studio
# Gift Cards

## Description

This PR adds Gift Card support to the new Cart page.

Known issues out of scope:
* The `PriceAdjustments` accordion resets back to the original set of sections that are open (`isOpen={true}`) after mutations occur.
* Applying an invalid card spits an error to the console

## Related Issue





Closes JIRA-78.

## Acceptance 



### Verification Stakeholders


@schensley

### Specification



### Verification Steps



Since Gift Cards is an enterprise edition feature, first set your `MAGENTO_BACKEND_EDITION` to `EE`. 

1. `yarn watch:all`
1. Go to the `/cart` page
1. Check the balance of a valid card: `02Y0LCHUNDRZ `.
1. Attempt to check the balance of an invalid card: `no_work`.
1. Apply a valid card: `02Y0LCHUNDRZ `.
1. Attempt to apply an invalid card: `no_work`.
1. Remove a card.
1. Simulate a remove card failure by applying a card, turning off your network, and clicking "remove". You should see an error message Toast appear.

Set your `MAGENTO_BACKEND_EDITION` back to `CE` (`CE` is also the default), and observe that the Cart page doesn't even show a Gift Card section in the Accordion.

## Screenshots / Screen Captures (if appropriate)

![Screen Shot 2020-01-29 at 11 19 54 AM](https://user-images.githubusercontent.com/13182778/73381073-c4a09600-428a-11ea-8d40-a79e8f2ce065.png)

## Checklist

* I have updated the documentation accordingly, if necessary.
* I have added tests to cover my changes, if necessary.
